### ISSUE TITLE
Color picker is gone after upgrade

### ISSUE THREAD

**Issue by ALeschinsky** [shutter-project/shutter#448]
#### Description of requested feature:

Previous version of shutter had a color picker feature which is extremely useful to me... can it be restored please?

**Comment by DarthGandalf** [comment]
Which versions are these (old, new)? What exact color picker are you talking about?

**Comment by ALeschinsky** [comment]
I don't have the old version installed anymore unfortunately... the color picker tool used to be in the color selection dialog inside the editor (from main window, press Edit, then click on Color at the bottom):

![image](https://user-images.githubusercontent.com/2109420/148142769-58baaa0f-7b32-48bd-b793-9e63adce8a92.png)

The new version in the screenshot doesn't have it... there was a dropper icon that looked somewhat like this:

![image](https://user-images.githubusercontent.com/2109420/148143095-38e2c031-5886-4ad4-b6f5-d9f807d39ff3.png)

The version I have ("new") is 0.98 Rev.1570.

**Comment by ALeschinsky** [comment]
I remember I had to install the Editor feature separately... sorry it was a while ago I don't remember all the details, but editor didn't work out of the box in that "old" version. But once installed, it had the color picker tool, which I miss dearly. Otherwise I love shutter, thank you so much for it :)

**Comment by Photon89** [comment]
It opens, when you press the plus button on the bottom left (just below "Custom" on your screenshot).

**Comment by ALeschinsky** [comment]
That's not it... "Choose color" dialog opens:

![image](https://user-images.githubusercontent.com/2109420/148228997-0ac65d27-f148-40dd-a590-e00ea8550419.png)

Pick color tool that I'm missing allowed to click on any pixel on the screen, and would choose the color of that pixel as either stroke or background color - extremely useful, and is now gone... Is there a way to install the old version where it was still available maybe?

**Comment by DarthGandalf** [comment]
This window looks slightly differently for me:
![Выделение_399](https://user-images.githubusercontent.com/325092/148230133-6bb72570-ec0a-490f-9ada-9406a00951aa.png)

See the button on the top left.

Though when I click it, it does nothing, hm

**Comment by Photon89** [comment]
I don't have such a button as well. In Arch repos we have GTK version 3.24.31, what about Gentoo? According to https://www.youtube.com/watch?v=0Sxlhq7m3LI this has been added in Gnome 3.30, don't know if this corresponds to GTK 3.30?

Arch: https://github.com/archlinux/svntogit-packages/blob/packages/gtk3/trunk/PKGBUILD
Gentoo: https://gitweb.gentoo.org/repo/gentoo.git/tree/x11-libs/gtk+/gtk+-2.24.33.ebuild

Maybe it is the "-D colord=yes" option in Arch which makes the difference? Though it is strange that a "yes" makes the picker be NOT there. :smiley:

**Comment by DarthGandalf** [comment]
I have x11-libs/gtk+-3.24.29

> Gnome 3.30, don't know if this corresponds to GTK 3.30?

Gnome version is not relevant to gtk version

**Comment by DarthGandalf** [comment]
I suppose gtk2 version of this widget could do something differently

**Comment by DarthGandalf** [comment]
Yes, gtk3 version of this widget is very different from what it was in gtk2: https://www.reddit.com/r/linux/comments/ejuwt2/i_want_to_vent_about_the_gtk3_color_picker_for_a/

Perhaps we can somehow make use of https://ubuntuhandbook.org/index.php/2020/08/gcolor3-gtk-3-color-picker-for-linux/

**Comment by Photon89** [comment]
Looks like the availability of the color picker in the GTK3 widget depends on whether X11 or Wayland is used, see this comment in the discussion you linked: https://www.reddit.com/r/linux/comments/ejuwt2/i_want_to_vent_about_the_gtk3_color_picker_for_a/fdbn5n7/?utm_source=reddit&utm_medium=web2x&context=3

> Perhaps we can somehow make use of https://ubuntuhandbook.org/index.php/2020/08/gcolor3-gtk-3-color-picker-for-linux/

Do they use a custom widget here? Gimp also has a more advanced one.
